code/script.js

```diff
@@ -222,104 +226,195 @@ const generateBoard = () => {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the person doesn't wear ${value}! Remove all people that wear ${value}`
       )
     }
   } else if (category === 'other') {
     // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      keepCharactersAccessoriesOther (category, value)
+      alert ( `Yes, the person is a ${value}! Keep all ${value}s`)
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      removeCharactersAccessoriesOther (category, value)
+      alert (`No, the person isn't a ${value}! Remove all ${value}s`)
     }
-  }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  } else if (category === 'hair') {
+  // Similar to the one above
+    if (keep) {
+    keepCharactersHairEyes (category, value)
+    alert ( `Yes, the person has ${value} hair! Keep all people with ${value} hair`)
+    } else {
+    removeCharactersHairEyes (category, value)
+    alert (`No, the person doesn't have ${value} hair! Remove all people with ${value} hair`)
+    }
+  
+  } else if (category === 'eyes') {
+    // Similar to the one above
+      if (keep) {
+      keepCharactersHairEyes (category, value)
+      alert ( `Yes, the person has ${value} eyes! Keep all people with ${value} eyes`)
+      } else {
+      removeCharactersHairEyes (category, value)
+      alert (`No, the person doesn't have ${value} eyes! Remove all people with ${value} eyes`)
+      }
+    }
+  // This function redraws the board with the remaining people.
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+
+// When clicking guess, the player first have to confirm that they want to make a guess. Store the interaction from the player in a variable.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+
+const userConfirmation = confirm("You sure? Please confirm!")
+if (userConfirmation == true) {
+   // If the player wants to guess, invoke the checkMyGuess function.
+  checkMyGuess(personToConfirm)
+} else {
```

Comment:
I would remove the else, since the else statement is empty i dont think it has a function.